packages/@aws-cdk/aws-events-targets/lib/kinesis-stream.ts

```diff
@@ -0,0 +1,63 @@
+import * as events from '@aws-cdk/aws-events';
+import * as iam from '@aws-cdk/aws-iam';
+import * as kinesis from '@aws-cdk/aws-kinesis';
+import { singletonEventRole } from './util';
+
+/**
+ * Customize the Kinesis Stream Event Target
+ */
+export interface KinesisStreamProps {
+  /**
+   * Partition Key Path for records sent to this stream
+   *
+   * @default eventId as the partition key
+   */
+  readonly partitionKeyPath?: string;
+
+  /**
+   * The message to send to the stream.
+   *
+   * Must be a valid JSON text passed to the target stream.
+   *
+   * @default the entire CloudWatch event
+   */
+  readonly message?: events.RuleTargetInput;
+
+}
+
+/**
+ * Use a Kinesis Stream as a target for AWS CloudWatch event rules.
+ *
+ * @example
+ *
+ *    // put to a Kinesis stream every time code is committed
+ *    // to a CodeCommit repository
+ *    repository.onCommit(new targets.KinesisStream(stream));
+ *
+ */
+export class KinesisStream implements events.IRuleTarget {
+
+  constructor(private readonly stream: kinesis.IStream, private readonly props: KinesisStreamProps = {}) {
+  }
+
+  /**
+   * Returns a RuleTarget that can be used to trigger this Kinesis Stream as a
+   * result from a CloudWatch event.
+   */
+  public bind(_rule: events.IRule, _id?: string): events.RuleTargetConfig {
+    const policyStatements = [new iam.PolicyStatement({
+      actions: ['kinesis:PutRecord', 'kinesis:PutRecords'],
+      resources: [this.stream.streamArn],
+    })];
+
+    return {
+      id: '',
```

Comment:
Does it make sense that the id is empty?

Reply:
I've mimicked what's already in other targets:
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/ecs-task.ts#L173
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/batch.ts#L68
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/codebuild.ts#L34
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/lambda.ts#L36
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/sns.ts#L42
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/sqs.ts#L65
* https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/state-machine.ts#L32

If it does not make sense, I would recommend to address this as separate pull request (and refactor all the existing targets accordingly).

Reply:
Okay, thanks